From 34205e38e60e426bc936df38a2089c9448f2b38e Mon Sep 17 00:00:00 2001
From: leventeliu <zhiteng.liu@covenantsql.io>
Date: Thu, 16 May 2019 11:28:04 +0800
Subject: [PATCH 1/2] Resolve unsupported FlagSet.Name() in go1.9

---
 cmd/cql/internal/help.go | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/cmd/cql/internal/help.go b/cmd/cql/internal/help.go
index 56b56361..e7caf577 100644
--- a/cmd/cql/internal/help.go
+++ b/cmd/cql/internal/help.go
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"os"
 	"runtime"
+	"strings"
 )
 
 const name = "cql"
@@ -76,10 +77,8 @@ func runVersion(cmd *Command, args []string) {
 	fmt.Print(PrintVersion(false))
 }
 
-func printParamHelp(flagSet *flag.FlagSet) {
-	if flagSet.Name() != "" {
-		_, _ = fmt.Fprintf(os.Stdout, "\n%s:\n", flagSet.Name())
-	}
+func printParamHelp(section string, flagSet *flag.FlagSet) {
+	_, _ = fmt.Fprintf(os.Stdout, "\n%s:\n", section)
 	flagSet.SetOutput(os.Stdout)
 	flagSet.PrintDefaults()
 }
@@ -89,13 +88,13 @@ func printCommandHelp(cmd *Command) {
 	_, _ = fmt.Fprintf(os.Stdout, cmd.Long)
 
 	if cmd.Flag != nil {
-		printParamHelp(cmd.Flag)
+		printParamHelp(fmt.Sprintf("%s params", strings.Title(cmd.Name())), cmd.Flag)
 	}
 	if cmd.CommonFlag != nil {
-		printParamHelp(cmd.CommonFlag)
+		printParamHelp("Common params", cmd.CommonFlag)
 	}
 	if cmd.DebugFlag != nil {
-		printParamHelp(cmd.DebugFlag)
+		printParamHelp("Debug params", cmd.DebugFlag)
 	}
 }
 
-- 
2.21.0

